### PR TITLE
Update to eslint 4 and update to blockly's new eslint config.

### PR DESCRIPTION
**Pull request by paulkaplan**

### CODE REVIEW

File: blocks_vertical/procedures.js [hunk 1]
```diff
@@ -185,7 +185,7 @@ Blockly.ScratchBlocks.ProcedureUtils.removeAllInputs_ = function() {
  */
 Blockly.ScratchBlocks.ProcedureUtils.createAllInputs_ = function(connectionMap) {
   // Split the proc into components, by %n, %b, and %s (ignoring escaped).
-  var procComponents = this.procCode_.split(/(?=[^\\]\%[nbs])/);
+  var procComponents = this.procCode_.split(/(?=[^\\]%[nbs])/);
```
Comment (kchadha) [inline blocks_vertical/procedures.js]: Was this just a bug before or is this an unintentional change...?

Comment (kchadha) [inline blocks_vertical/procedures.js:188]: Is this actually a lint related change?

File: blocks_vertical/vertical_extensions.js [hunk 2]
```diff
@@ -223,10 +223,9 @@ Blockly.ScratchBlocks.VerticalExtensions.SCRATCH_EXTENSION = function() {
 Blockly.ScratchBlocks.VerticalExtensions.registerAll = function() {
   var categoryNames =
       ['control', 'data', 'data_lists', 'sounds', 'motion', 'looks', 'event',
-      'sensing', 'pen', 'operators', 'more'];
+        'sensing', 'pen', 'operators', 'more'];
   // Register functions for all category colours.
-  for (var i = 0; i < categoryNames.length; i++) {
-    name = categoryNames[i];
```
Comment (kchadha) [inline blocks_vertical/vertical_extensions.js:229]: Where was `name` being declared before?

File: core/block.js [hunk 3]
```diff
@@ -1196,7 +1196,7 @@ Blockly.Block.prototype.jsonInit = function(json) {
   // Validate inputs.
   goog.asserts.assert(json['output'] == undefined ||
       json['previousStatement'] == undefined,
-      'Must not have both an output and a previousStatement.');
+  'Must not have both an output and a previousStatement.');
```
Comment (kchadha) [inline core/block.js:1199]: This seems like the opposite of the other changes...? This is still a line continuation, right?

File: core/comment.js [hunk 4]
```diff
@@ -70,21 +70,21 @@ Blockly.Comment.prototype.drawIcon_ = function(group) {
...
       {'class': 'blocklyIconShape', 'r': '8', 'cx': '8', 'cy': '8'},
-       group);
+      group);
   // Can't use a real '?' text character since different browsers and operating
   // systems render it differently.
   // Body of question mark.
   Blockly.utils.createSvgElement('path',
       {'class': 'blocklyIconSymbol',
-       'd': 'm6.8,10h2c0.003,-0.617 0.271,-0.962 0.633,-1.266 2.875,-2.405' +
+        'd': 'm6.8,10h2c0.003,-0.617 0.271,-0.962 0.633,-1.266 2.875,-2.405' +
```
Comment (kchadha) [inline core/comment.js:79]: Shouldn't the lines below this also get indented to match up with this line?

File: core/workspace_svg.js [hunk 5]
```diff
@@ -423,7 +423,7 @@ Blockly.WorkspaceSvg.prototype.createDom = function(opt_backgroundClass) {
     bottom = this.addTrashcan_(bottom);
   }
   if (this.options.zoomOptions && this.options.zoomOptions.controls) {
-    bottom = this.addZoomControls_(bottom);
+    this.addZoomControls_(bottom);
```
Comment (kchadha) [inline core/workspace_svg.js:426]: why this change?

File: core/xml.js [hunk 6]
```diff
@@ -623,13 +623,12 @@ Blockly.Xml.domToBlockHeadless_ = function(xmlBlock, workspace) {
...
-        if (grandchildNode.nodeName.toLowerCase() == 'block') {
-          childBlockNode = grandchildNode;
-        } else if (grandchildNode.nodeName.toLowerCase() == 'shadow') {
-          childShadowNode = grandchildNode;
+    for (var j = 0, grandchild; grandchild = xmlChild.childNodes[j]; j++) {
+      if (grandchild.nodeType == 1) {
+        if (grandchild.nodeName.toLowerCase() == 'block') {
+          childBlockNode = grandchild;
+        } else if (grandchild.nodeName.toLowerCase() == 'shadow') {
+          childShadowNode = grandchild;
```
Comment (kchadha) [inline core/xml.js:631]: I remember we made this name change (and probably others above) intentionally, but I can't remember why...

File: core/comment.js (hunk 4 above)
Comment (kchadha) [reply core/comment.js:79]: I revise my statement, the lines below should get indented 2 further than line 79 above

**Comment by rachel-fenichel** [comment]
Per previous discussion, merging this will cause merge problems with blockly--there are multiple ways to solve some of the lint issues, and we did them different ways.  It's a silly problem, but so are most lint problems.  I'm incrementally fixing lint as I do the merge right now.

**Comment by kchadha** [comment]
Closing this PR as the changes were addressed in #1479
